#!/usr/bin/env python
PACKAGE = "exploration_support"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("obstacle_check_angle", double_t, 0, "", 0.04, 0.0, 1.57)
gen.add("obstacle_range_threshold", double_t, 0, "", 2.5, 0.0, 10.0)
gen.add("branch_range_max", double_t, 0, "", 6.0, 0.0, 10.0)
gen.add("branch_range_min", double_t, 0, "", 0.3, 0.0, 10.0)
gen.add("branch_diff_x_min", double_t, 0, "", 1.0, 0.0, 10.0)
gen.add("branch_diff_x_max", double_t, 0, "", 10.0, 0.0, 10.0)
gen.add("branch_diff_y_min", double_t, 0, "", 0.0, 0.0, 10.0)
gen.add("branch_diff_y_max", double_t, 0, "", 10.0, 0.0, 10.0)
gen.add("scan_filter", bool_t, 0, "", False)
gen.add("scan_filter_order", int_t, 0, "", 3, 1, 10)
gen.add("branch_filter", bool_t, 0, "", False)
gen.add("branch_filter_order", int_t, 0, "", 3, 1, 10)
gen.add("branch_filter_tolerance", double_t, 0, "", 0.2, 0.0, 10.0)
gen.add("duplicate_detection", bool_t, 0, "", True)
gen.add("duplicate_tolerance", double_t, 0, "", 1.5, 0, 10.0)
gen.add("log_current_time", double_t, 0, "", 10.0, 0.0, 600.0)
gen.add("newer_duplication_threshold", double_t, 0, "", 100.0, 0.0, 1000000.0)
gen.add("on_map_branch_detection", bool_t, 0, "", True)
gen.add("omb_map_window_x", double_t, 0, "", 1.0, 0.0, 10.0)
gen.add("omb_map_window_y", double_t, 0, "", 1.0, 0.0, 10.0)
gen.add("on_map_branch_rate", double_t, 0, "", 0.5, 0.0, 1.0)


exit(gen.generate(PACKAGE, "exploration_support", "branch_detection_parameter_reconfigure"))